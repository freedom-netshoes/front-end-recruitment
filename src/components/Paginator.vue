<template>
  <div class="paginator">
    <div class="previous">
      anterior
    </div>
    <div 
      v-for="page in pages"
      :class="{ 'page': true, 'current' : isPageCurrent(page) }" 
      @click="paginate(page)"
    >
      {{ page }}
      <div class="separator"/>
    </div>
    <div class="separator-right"/>
    <div class="next">
      pr√≥ximo
    </div>
  </div>
</template>
<script>
export default {
  name: 'Paginator',
  data: function () {
    return {
      pages: [1, 2, 3, 4],
      currentPage: 1
    }
  },
  methods: {
    isPageCurrent: function (page) {
      return this.currentPage === page
    },
    paginate: function (page) {
      this.currentPage = page
    }
  }
}
</script>
<style scoped lang="scss">
  .paginator {
    height: 48px;
    max-width: 440px;
    border: 2px solid $grey;
    text-transform: uppercase;
    display: flex;
    align-items: center;
    margin-bottom: $spacing-lg;
    border-radius: 4px;
    font-weight: bold;
    color: $grey;
    position: relative;

    .separator-right {
      height: 10px;
      background: $grey;
      width: 2px;
    }
  
    .page {
      padding: 0 $spacing-sm;
      height: 100%;
      display: flex;
      align-items: center;
      flex-direction: column;
      justify-content: center;

      &:hover {
        color: $color-secondary;
        cursor: pointer;
      }

      &.current {
        color: $yellow;

        .separator {
          height: 2px;
          background: $yellow;
          width: 14px;
        }
      }
    }

    .previous,
    .next {
      margin: 0 $spacing-md;

      &:hover {
        color: $color-secondary;
        cursor: pointer;
      }
    }
  }
</style>
