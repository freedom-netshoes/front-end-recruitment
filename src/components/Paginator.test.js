import { shallow } from 'vue-test-utils'
import Paginator from './Paginator'

describe('Paginator.test.js', () => {
  let cmp

  beforeEach(() => {
    cmp = shallow(Paginator, {
      data: {
        pages: [1, 2, 3, 4, 5, 6, 7]
      }
    })
  })

  it('equals messages to [1, 2, 3, 4, 5, 6, 7]', () => {
    expect(cmp.vm.pages).toEqual([1, 2, 3, 4, 5, 6, 7])
  })

  it('has the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot()
  })
})