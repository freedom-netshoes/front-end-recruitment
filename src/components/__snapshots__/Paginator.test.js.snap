// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator.test.js has the expected html structure 1`] = `
<div
  class="paginator"
>
  <div
    class="previous"
  >
    
    anterior
  
  </div>
   
  <div
    class="pages-container"
  >
    <div
      class="page current"
    >
      
      1
      
      <div
        class="separator"
      />
    </div>
     
    <div
      class="separator-right"
    />
  </div>
  <div
    class="pages-container"
  >
    <div
      class="page"
    >
      
      2
      
      <div
        class="separator"
      />
    </div>
     
    <div
      class="separator-right"
    />
  </div>
  <div
    class="pages-container"
  >
    <div
      class="page"
    >
      
      3
      
      <div
        class="separator"
      />
    </div>
     
    <div
      class="separator-right"
    />
  </div>
  <div
    class="pages-container"
  >
    <div
      class="page"
    >
      
      4
      
      <div
        class="separator"
      />
    </div>
     
    <div
      class="separator-right"
    />
  </div>
  <div
    class="pages-container"
  >
    <div
      class="page"
    >
      
      5
      
      <div
        class="separator"
      />
    </div>
     
    <div
      class="separator-right"
    />
  </div>
  <div
    class="pages-container"
  >
    <div
      class="page"
    >
      
      6
      
      <div
        class="separator"
      />
    </div>
     
    <div
      class="separator-right"
    />
  </div>
  <div
    class="pages-container"
  >
    <div
      class="page"
    >
      
      7
      
      <div
        class="separator"
      />
    </div>
     
    <!---->
  </div>
   
  <div
    class="next"
  >
    
    pr√≥ximo
  
  </div>
</div>
`;
